lib/rust/ensogl/core/src/display/navigation/navigator.rs

```diff
@@ -29,25 +41,34 @@ pub struct NavigatorModel {
     /// Indicates whether events handled the navigator should be stopped from propagating further
     /// after being handled by the Navigator.
     disable_events:  Rc<Cell<bool>>,
+    max_zoom_limit:  Rc<Cell<Option<f32>>>,
 }
 
 impl NavigatorModel {
     pub fn new(scene: &Scene, camera: &Camera2d) -> Self {
         let zoom_speed = Rc::new(Cell::new(Switch::On(10.0 / 1000.0)));
         let pan_speed = Rc::new(Cell::new(Switch::On(1.0)));
-        let min_zoom = 10.0;
-        let max_zoom = 10000.0;
         let disable_events = Rc::new(Cell::new(true));
+        let max_zoom_limit = default();
         let (simulator, resize_callback, _events) = Self::start_navigator_events(
             scene,
             camera,
-            min_zoom,
-            max_zoom,
+            MIN_ZOOM,
+            MAX_ZOOM,
             Rc::clone(&zoom_speed),
             Rc::clone(&pan_speed),
             Rc::clone(&disable_events),
+            Rc::clone(&max_zoom_limit),
```

Comment:
Why we use `Rc::clone(` notations instead of just `.clone()`?

Reply:
To clearly disambiguate cloning of pointer from cloning of the underlying data. It's a pretty common pattern in the Rust codebases I've worked with: https://rust-lang.github.io/rust-clippy/master/#clone_on_ref_ptr

Reply:
We are using `clone_ref` here instead everywhere to show the same thing :)

Reply:
@vitvakatu absolutely! Also, `clone_ref` was not always available, so some old code can use different patterns. Sorry if my message was not clear enough – I wanted t write that nowadays we normally use `clone_ref` :)

Reply:
Changed